pkg/types/dsse/v0.0.1/entry_test.go

```diff
@@ -0,0 +1,509 @@
+			SignatureSigner: s,
+		})
+	}
+
+	signer, err := dsse.NewMultiEnvelopeSigner(2, evps[0], evps[1])
+	if err != nil {
+		t.Fatal(err)
+	}
+	dsseEnv, err := signer.SignPayload(context.Background(), in_toto.PayloadType, payload)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	return dsseEnv
+}
+
+func createRekorEnvelope(dsseEnv *dsse.Envelope, pub [][]byte) *models.DSSEV001SchemaProposedContent {
+
+	envelopeBytes, _ := json.Marshal(dsseEnv)
+	proposedContent := &models.DSSEV001SchemaProposedContent{
+		Envelope: swag.String(string(envelopeBytes)),
+	}
+	for _, key := range pub {
+		proposedContent.PublicKeys = append(proposedContent.PublicKeys, strfmt.Base64(key))
+	}
+	return proposedContent
+}
+
+func TestV001Entry_Unmarshal(t *testing.T) {
+	key, err := ecdsa.GenerateKey(elliptic.P256(), rand.Reader)
+	if err != nil {
+		t.Fatal(err)
+	}
+	der, err := x509.MarshalPKIXPublicKey(&key.PublicKey)
+	if err != nil {
+		t.Fatal(err)
+	}
+	pub := pem.EncodeToMemory(&pem.Block{
+		Bytes: der,
+		Type:  "PUBLIC KEY",
+	})
+
+	priv, err := ecdsa.GenerateKey(elliptic.P256(), rand.Reader)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	ca := &x509.Certificate{
+		SerialNumber: big.NewInt(1),
+	}
+	caBytes, err := x509.CreateCertificate(rand.Reader, ca, ca, &priv.PublicKey, priv)
+	if err != nil {
+		t.Fatal(err)
+	}
+	pemBytes := pem.EncodeToMemory(&pem.Block{
+		Type:  "CERTIFICATE",
+		Bytes: caBytes,
+	})
+
+	invalid := dsse.Envelope{
+		Payload: "hello",
+		Signatures: []dsse.Signature{
+			{
+				Sig: string(strfmt.Base64("foobar")),
+			},
+		},
+	}
+
+	validPayload := "hellothispayloadisvalid"
+
+	tests := []struct {
+		env     *dsse.Envelope
+		name    string
+		it      *models.DSSEV001Schema
+		wantErr bool
+	}{
+		{
+			name:    "empty",
+			it:      &models.DSSEV001Schema{},
+			wantErr: true,
+		},
+		{
+			name: "missing envelope",
+			it: &models.DSSEV001Schema{
+				ProposedContent: &models.DSSEV001SchemaProposedContent{},
+			},
+			wantErr: true,
+		},
+		{
+			env:  envelope(t, key, []byte(validPayload)),
+			name: "valid",
+			it: &models.DSSEV001Schema{
+				ProposedContent: createRekorEnvelope(envelope(t, key, []byte(validPayload)), [][]byte{pub}),
+			},
+			wantErr: false,
+		},
+		{
+			env:  envelope(t, priv, []byte(validPayload)),
+			name: "cert",
+			it: &models.DSSEV001Schema{
+				ProposedContent: createRekorEnvelope(envelope(t, priv, []byte(validPayload)), [][]byte{pemBytes}),
+			},
+			wantErr: false,
+		},
+		{
+			env:  &invalid,
+			name: "invalid",
+			it: &models.DSSEV001Schema{
+				ProposedContent: createRekorEnvelope(&invalid, [][]byte{pub}),
+			},
+			wantErr: true,
+		},
+		{
+			env:  envelope(t, key, []byte(validPayload)),
+			name: "invalid key",
+			it: &models.DSSEV001Schema{
+				ProposedContent: createRekorEnvelope(envelope(t, key, []byte(validPayload)), [][]byte{[]byte("notavalidkey")}),
+			},
+			wantErr: true,
+		},
+		{
+			env:  multiSignEnvelope(t, []*ecdsa.PrivateKey{key, priv}, []byte(validPayload)),
```

Comment:
Do we have a test for a mismatch between signatures and public keys, that not all keys can verify all signatures?

Reply:
those are in e2e tests